services/roku-channel-store/roku-channel-store-base.js

```diff
@@ -0,0 +1,30 @@
+import Joi from 'joi'
+import { nonNegativeInteger } from '../validators.js'
+import { BaseJsonService } from '../index.js'
+
+const schema = Joi.object({
+  rating: Joi.number().min(0),
+  starRating: Joi.number().min(0),
+  starRatingCount: nonNegativeInteger,
+}).required()
+
+export default class RokuChannelStoreBase extends BaseJsonService {
+  async fetch({ channelId }) {
+    const url = `https://channelstore.roku.com/api/v6/channels/${channelId}`
```

Comment:
Is there any documentation for this API?

Also can you explain what is sold in the Roku Channel store? It looks like it is more like TV and radio content than software.

Reply:
> Is there any documentation for this API?

Not that I know of

> Also can you explain what is sold in the Roku Channel store? It looks like it is more like TV and radio content than software.

They are tailored towards TV (since Roku is a TV OS) but it's still apps.
Here's a couple of example repos that could display the badges
https://github.com/iBicha/playlet (store link https://channelstore.roku.com/en-ca/details/840aec36f51bfe6d96cf6db9055a372a/playlet)
https://github.com/jellyfin/jellyfin-roku (store link https://channelstore.roku.com/en-ca/details/cc5e559d08d9ec87c5f30dcebdeebc12/jellyfin)

Reply:
OK thanks. The reason we say

> Badges should not obtain data from undocumented or reverse-engineered API endpoints.

in [our guidelines](https://github.com/badges/shields/blob/master/CONTRIBUTING.md#badge-guidelines)

is because we had some situations where a contributor worked out that a project had an API by inspecting the frontend code for their website or whatever, added a badge that calls it, and then we ended up relying on an API that wasn't really intended for public use. This then either annoyed the maintainers of the company/project behind it or we had no reasonable expectation of stability/compatibility over time (because devs maintaining the the API were making the assumption the only thing consuming it was their own frontend code) and stuff kept breaking.

That's looks like the situation we've got here so I am going to close this PR, but we could revisit in future if there is a reliable public source for the data.